main/api/src/mill/api/Ctx.scala

```diff
@@ -61,7 +60,7 @@ class Ctx(
   val log: Logger,
   val home: os.Path,
   val env: Map[String, String],
-  val reporter: Int => Option[ManagedLoggedReporter],
+  val reporter: Int => Option[ZincProblemReporter],
```

Comment:
@adadima can you please explain why the Int to Option function was required here. What is the key in this case and why it isn't just Option ?

Reply:
Each Build Target needs a separate reporter to be injected in the compilation and  the construction of the Reporter depends on some information about the BuildTarget being compiled. Since the reporter is generated at some point in the evaluate() method on Evaluator ( which has no knowledge of bsp build targets ), I needed some piece of information accessible from Evaluator that can be converted to a BuildTarget inside MillBuildServer: that is the hash code of the mill module associated with the Build Target. I should  find a more reliable way of doing this than hash code, but I didn't have a chance to yet.